packages/mui-joy/src/styles/CssVarsProvider.tsx

```diff
@@ -30,6 +33,10 @@ const { CssVarsProvider, useColorScheme, getInitColorSchemeScript } = createCssV
       mergedTheme.colorInversion,
       { clone: false },
     );
+
+    mergedTheme.unstable_sx = function sx(props: SxProps) {
+      return styleFunctionSx({ sx: props, theme: this });
+    };
```

Comment:
As we move forward, I think that we could save this intermediary function call and object allocation with:

```suggestion
    mergedTheme.unstable_sx = styleFunctionSx(mergedTheme);
```

Until then, we can leave a TODO
```suggestion
    // TOOD remove this intermediary function call.
    mergedTheme.unstable_sx = function sx(props: SxProps) {
      return styleFunctionSx({ sx: props, theme: this });
    };
```